crates/egui_extras/src/image.rs

```diff
@@ -279,3 +301,25 @@ pub fn load_svg_bytes_with_size(
 
     Ok(image)
 }
+
+#[cfg(feature = "image")]
+pub fn convert_image_to_bytes(image: &image::DynamicImage) -> Bytes {
+    let image_buffer = image.to_rgba8();
+    let pixels = image_buffer.as_flat_samples();
+    let mut result: Vec<u8> = Vec::new();
+    result.extend_from_slice(&(image.width() as usize).to_le_bytes());
+    result.extend_from_slice(&(image.height() as usize).to_le_bytes());
+    result.extend_from_slice(pixels.as_slice());
+    result.into()
+}
```

Comment:
A simpler solution here is to encode it as a lossless, uncompressed `.tga`. Then `egui_extras::ImageLoader` will load it for you (no need to write an extra loader!).

Reply:
my problem with that is that I encode something to decode it right after

Reply:
Ok, but then I suggest you add the decoding function here, next to the encoding function, and that you at least add some magic header to it as a sanity check